src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -163,7 +163,7 @@ class ImplLookup(private val project: Project, private val items: StdKnownItems)
         return subst + associated
     }
 
-    fun derefSequence(baseTy: Ty): Sequence<Ty> {
+    fun coercionSequence(baseTy: Ty): Sequence<Ty> {
```

Comment:
Is this intended to include real coercions in the future? For the current implementation, `deref` makes more sense to me as a name...

Reply:
This method already does coercion because it converts `TyArray` to `TySlice`.
But if you consider the previous name is better I'll revert renaming - I don't insist:)

Reply:
Let's make it `coercion` then, @vlad20012 could change it later if he disagrees :)